### DISCUSSION TITLE
Is it possible to highlight the PDF downloaded from box.com and write it to another HttpResponse stream on the fly(without Disk I/O)?

### DISCUSSION THREAD

**Discussion by mingchun-zhao** [pymupdf/PyMuPDF#1906] [Q&A] [answered]
Hi there,

Is it possible to highlight the PDF obtained from Box and write that result to another HttpResponse stream on the fly?
Anyway, I don't want to cause disk reads and writes.

Any advice would be appreciated.

**Comment by JorjMcKie** [comment, marked as answer]
What does that mean: "obtained from BOX"?

**Reply by mingchun-zhao** [reply]
Thank you. It means to get files from cloud storage box.com using API.
Sorry for the confusion, I've changed the title.

**Reply by JorjMcKie** [reply]
PyMuPDF does not support dwonloading / uploading to / from the internet.
But once you have a memory version of a PDF from whatever source, you can do everything with it as with a disk-resident file.
When finished (e.g. after highlighting whatever stuff on whatever page), create a new memory image from it via `doc.tobytes( ...whatever parameters ...)` and deliver it (a `bytes` object) to some place (e.g. upload it).

**Reply by mingchun-zhao** [reply]
got it, thank you so much.
